File: dev/test_data/python/test_main/aas_core_meta.v3/input/snippets/Verification/lang_strings_have_unique_languages.py
def lang_strings_have_unique_languages(
    lang_strings: Iterable[aas_types.AbstractLangString],
) -> bool:
    """
    Check that :paramref:`lang_strings` are specified each for a unique
    language.
    """
    language_set = set()  # type: Set[str]
    for lang_string in lang_strings:
        if lang_string.language in language_set:
            return False

        language_set.add(lang_string.language)

    return True

File: dev/test_data/csharp/test_main/aas_core_meta.v3/input/snippets/Verification/lang_strings_have_unique_languages.cs
/// <summary>
/// Check that <paramref name="langStrings" /> are specified each for a unique
/// language.
/// </summary>
public static bool LangStringsHaveUniqueLanguages(
    IEnumerable<Aas.IAbstractLangString> langStrings
)
{
    var languageSet = new HashSet<string>();
    foreach (var langString in langStrings)
    {
        if (languageSet.Contains(langString.Language))
        {
            return false;
        }
        languageSet.Add(langString.Language);
    }
    return true;
}
